src/executables/util_cs.c

```diff
@@ -4000,6 +4000,58 @@ tde (UTIL_FUNCTION_ARG * arg)
   return EXIT_FAILURE;
 }
 
+static int
+check_table_name (const char *table_name)
+{
+  int table_name_len = STATIC_CAST (int, strlen (table_name));
+  int sub_len = 0;
+  const char *dot = NULL;
+
+  if (table_name_len >= SM_MAX_IDENTIFIER_LENGTH)
+    {
+      PRINT_AND_LOG_ERR_MSG (msgcat_message
+			     (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_FLASHBACK,
+			      FLASHBACK_MSG_EXCEED_MAX_OWNER_CLASS_LENGTH), SM_MAX_IDENTIFIER_LENGTH);
+
+      return ER_FAILED;
+    }
+
+  dot = strchr (table_name, '.');
+  if (dot == NULL)
+    {
+      /* owner name is not specified */
+      PRINT_AND_LOG_ERR_MSG (msgcat_message
+			     (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_FLASHBACK, FLASHBACK_MSG_OWNER_NOT_SPECIFIED),
+			     table_name);
+
+      return ER_FAILED;
+    }
+
+  /* check length of owner name */
+  sub_len = STATIC_CAST (int, dot - table_name);
+  if (sub_len < 1 || sub_len >= SM_MAX_USER_LENGTH)
+    {
+      PRINT_AND_LOG_ERR_MSG (msgcat_message
+			     (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_FLASHBACK, FLASHBACK_MSG_INVALID_OWNER_LENGTH),
+			     SM_MAX_USER_LENGTH);
+
+      return ER_FAILED;
+    }
+
+  /* check length of class name */
+  sub_len = STATIC_CAST (int, strlen (dot + 1));
+  if (sub_len < 1 || sub_len >= SM_MAX_IDENTIFIER_LENGTH - SM_MAX_USER_LENGTH)
```

Comment:
What is the maximum length of the class name in CUBRID? If it is SM_MAX_IDENTIFIER_LENGTH, this requirement prevents the flashback if the length of a user name + a length of a class name is bigger than SM_MAX_IDENTIFIER_LENGTH.

If so, the max length of owner_name.class_name, `table_name_len`,  should be SM_MAX_IDENTIFIER_LENGTH + SM_MAX_USER_LENGTH.

Reply:
What has changed when user schema is applied is this length spec (`SM_MAX_IDENTIFIER_LENGTH`).
SM_MAX_IDENTIFIER_LEGNTH represents maximum value of (length of owner name + length of class name).

So, I compared class name length with (SM_MAX_IDENTIFIER_LENGTH - SM_MAX_USER_LENGTH)

Reply:
OK, I got it. Then, what about the backward compatibility? Do the existing table and user name have to be changed to fit that restriction? I don't think it's a flashback's issue, but I just wonder.

Reply:
Yes. As I know, the functions that check the length of the class name and owner name have been changed according to the specification. (`get_class_mops_from_file()`, `get_class_mops()`, `get_requested_classes()`,.. )

`SM_MAX_IDENTIFIER_LENGTH` is used for unique_name (owner_name.class_name), and unique name is used in server side instead of class name through parser where makes classname to unique_name.

Reply:
But, I don't know about the existing tables. I agree with you, it can cause a problem.

Reply:
Thank you for the answer 👍 .